cfgrammar/src/lib/yacc/grammar.rs

```diff
@@ -45,6 +45,12 @@ pub struct YaccGrammar<StorageT = u32> {
     /// A mapping from `TIdx` -> `Option<String>`. Every user-specified token will have a name,
     /// but tokens inserted by cfgrammar (e.g. the EOF token) won't.
     token_names: Vec<Option<String>>,
+    /// A mapping from `TIdx` -> `Option<(usize, usize)>`.
+    /// The first element is the absolute byte offset of the first element of the name
+    /// in the yacc file this grammar was produced from (if any).
+    /// The second is the absolute byte offset of the end of the name.
+    /// Elements not represented by a yacc source file, or inserted by cfgrammar return None.
```

Comment:
Should we just have `token_names: Vec<Option<(Span, String)>>` rather than two separate, but highly related, fields?

Reply:
Will give that a go!

Reply:
I haven't had much luck with this, It's easy enough to build the `Vec<Option<(Span, String)>>`,
but all of the `name_option.as_ref()` have given me a bit of trouble, (I think it is tuple not implementing as_ref).
And `x.map(|(_, x)| x).as_ref()` as giving me some inference troubles,

But i'll try it again now that I'm not trying to change 2 things at once.

In theory I agree `Vec<Option<(Span, String)>>` is nicer, just not sure implementation wise yet.

Reply:
Not so bad if you don't accidentally mix up `(Span, String)` with ``(String, Span)`.